diff --git a/arch/arm/boot/dts/stm32mp157d-atk.dts b/arch/arm/boot/dts/stm32mp157d-atk.dts
index a4f922f35..906af5e32 100644
--- a/arch/arm/boot/dts/stm32mp157d-atk.dts
+++ b/arch/arm/boot/dts/stm32mp157d-atk.dts
@@ -37,6 +37,15 @@
 			no-map;
 		};
 	};
+
+	key {
+		compatible = "alientek,key";
+		status = "okay";
+		key-gpio = <&gpiog 3 GPIO_ACTIVE_LOW>;
+		interrupt-parent = <&gpiog>;
+		interrupts = <3 IRQ_TYPE_EDGE_BOTH>;
+	};
+
 };
 
 &cpu1 {
diff --git a/arch/arm/boot/dts/stm32mp157d-atk.dtsi b/arch/arm/boot/dts/stm32mp157d-atk.dtsi
index b11f359cf..03e383bdb 100644
--- a/arch/arm/boot/dts/stm32mp157d-atk.dtsi
+++ b/arch/arm/boot/dts/stm32mp157d-atk.dtsi
@@ -80,13 +80,14 @@
 	gpio-keys {
 			compatible = "gpio-keys";
 			autorepeat;
+/*
 			key0 {
 				label = "USER-KEY0";
 				linux,code = <114>;
 				gpios = <&gpiog 3 GPIO_ACTIVE_LOW>;
 				gpio-key,wakeup;
 			};
-
+*/
 			key1 {
 				label = "USER-KEY1";
 				linux,code = <115>;
