diff --git a/arch/arm/boot/dts/stm32mp15-pinctrl.dtsi b/arch/arm/boot/dts/stm32mp15-pinctrl.dtsi
index 36624234e..5013a5b14 100644
--- a/arch/arm/boot/dts/stm32mp15-pinctrl.dtsi
+++ b/arch/arm/boot/dts/stm32mp15-pinctrl.dtsi
@@ -553,9 +553,7 @@
 
 	pwm1_pins_a: pwm1-0 {
 		pins {
-			pinmux = <STM32_PINMUX('E', 9, AF1)>, /* TIM1_CH1 */
-				 <STM32_PINMUX('E', 11, AF1)>, /* TIM1_CH2 */
-				 <STM32_PINMUX('E', 14, AF1)>; /* TIM1_CH4 */
+			pinmux = <STM32_PINMUX('A', 10, AF1)>; /* TIM1_CH3 */
 			bias-pull-down;
 			drive-push-pull;
 			slew-rate = <0>;
@@ -564,9 +562,7 @@
 
 	pwm1_sleep_pins_a: pwm1-sleep-0 {
 		pins {
-			pinmux = <STM32_PINMUX('E', 9, ANALOG)>, /* TIM1_CH1 */
-				 <STM32_PINMUX('E', 11, ANALOG)>, /* TIM1_CH2 */
-				 <STM32_PINMUX('E', 14, ANALOG)>; /* TIM1_CH4 */
+			pinmux = <STM32_PINMUX('A', 10, ANALOG)>; /* TIM1_CH3 */
 		};
 	};
 
@@ -1341,7 +1337,7 @@
 
 	usbotg_hs_pins_a: usbotg_hs-0 {
 		pins {
-			pinmux = <STM32_PINMUX('A', 10, ANALOG)>; /* OTG_ID */
+			//pinmux = <STM32_PINMUX('A', 10, ANALOG)>; /* OTG_ID */
 		};
 	};
 
diff --git a/arch/arm/boot/dts/stm32mp157-m4-srm-pinctrl.dtsi b/arch/arm/boot/dts/stm32mp157-m4-srm-pinctrl.dtsi
index b4030e5c9..b944b6d02 100644
--- a/arch/arm/boot/dts/stm32mp157-m4-srm-pinctrl.dtsi
+++ b/arch/arm/boot/dts/stm32mp157-m4-srm-pinctrl.dtsi
@@ -494,7 +494,7 @@
 
 	m4_usbotg_hs_pins_a: m4-usbotg_hs-0 {
 		pins {
-			pinmux = <STM32_PINMUX('A', 10, RSVD)>; /* OTG_ID */
+			//pinmux = <STM32_PINMUX('A', 10, RSVD)>; /* OTG_ID */
 		};
 	};
 
diff --git a/arch/arm/boot/dts/stm32mp157d-atk.dts b/arch/arm/boot/dts/stm32mp157d-atk.dts
index a4f922f35..cabaa1ad6 100644
--- a/arch/arm/boot/dts/stm32mp157d-atk.dts
+++ b/arch/arm/boot/dts/stm32mp157d-atk.dts
@@ -51,3 +51,17 @@
 &optee {
 	status = "okay";
 };
+
+&timers1 {
+	status = "okay";
+	/* spare all DMA channels since they are not needed for PWM output */
+	/delete-property/dmas;
+	/delete-property/dma-names;
+	pwm1: pwm {
+		pinctrl-0 = <&pwm1_pins_a>;
+		pinctrl-1 = <&pwm1_sleep_pins_a>;
+		pinctrl-names = "default", "sleep";
+		#pwm-cells = <2>;
+		status = "okay";
+	};
+};
\ No newline at end of file
diff --git a/arch/arm/boot/dts/stm32mp157d-atk.dtsi b/arch/arm/boot/dts/stm32mp157d-atk.dtsi
index b11f359cf..5917bd15f 100644
--- a/arch/arm/boot/dts/stm32mp157d-atk.dtsi
+++ b/arch/arm/boot/dts/stm32mp157d-atk.dtsi
@@ -448,6 +448,7 @@
 				<STM32_PINMUX('B', 12, ANALOG)>; /* UART5_RX */
 	   };
    };
+#if 0
 	pwm1_pins_a: pwm1-0 {
 		pins {
 			pinmux = <STM32_PINMUX('A', 10, AF1)>; /* TIM1_CH3 */
@@ -462,7 +463,7 @@
 			pinmux = <STM32_PINMUX('A', 10, ANALOG)>; /* TIM1_CH3 */
 		};
 	};
-
+#endif
    i2s2_pins_b: i2s2-2 {
 	   pins {
 		   pinmux = <STM32_PINMUX('C', 3, AF5)>, /* I2S2_SDO */
@@ -1024,7 +1025,7 @@
 };
 
 &timers4 {
-	status = "okay";
+	status = "disable";
 	/* spare dmas for other usage */
 	/delete-property/dmas;
 	/delete-property/dma-names;
@@ -1033,7 +1034,7 @@
 		pinctrl-1 = <&pwm4_sleep_pins_b>;
 		pinctrl-names = "default", "sleep";
 		#pwm-cells = <2>;
-		status = "okay";
+		status = "disable";
 	};
 };
 
diff --git a/arch/arm/boot/dts/stm32mp15xx-dkx.dtsi b/arch/arm/boot/dts/stm32mp15xx-dkx.dtsi
index 35169385f..21f69a0cc 100644
--- a/arch/arm/boot/dts/stm32mp15xx-dkx.dtsi
+++ b/arch/arm/boot/dts/stm32mp15xx-dkx.dtsi
@@ -198,7 +198,7 @@
 		reg = <0x39>;
 		iovcc-supply = <&v3v3_hdmi>;
 		cvcc12-supply = <&v1v2_hdmi>;
-		reset-gpios = <&gpioa 10 GPIO_ACTIVE_LOW>;
+		//reset-gpios = <&gpioa 10 GPIO_ACTIVE_LOW>;
 		interrupts = <1 IRQ_TYPE_EDGE_FALLING>;
 		interrupt-parent = <&gpiog>;
 		#sound-dai-cells = <0>;
@@ -649,13 +649,13 @@
 	/delete-property/dma-names;
 	status = "disabled";
 	pwm {
-		pinctrl-0 = <&pwm4_pins_a &pwm4_pins_b>;
-		pinctrl-1 = <&pwm4_sleep_pins_a &pwm4_sleep_pins_b>;
+		//pinctrl-0 = <&pwm4_pins_a &pwm4_pins_b>;
+		//pinctrl-1 = <&pwm4_sleep_pins_a &pwm4_sleep_pins_b>;
 		pinctrl-names = "default", "sleep";
-		status = "okay";
+		status = "disable";
 	};
 	timer@3 {
-		status = "okay";
+		status = "disable";
 	};
 };
 
